src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -155,7 +165,7 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
                 currentBuildResultFailure(build, listener, previousBuildResult, filename, vars);
             }
 
-            if (currentBuildResult == Result.SUCCESS) {
+            if (currentBuildResult == Result.SUCCESS && !newIssue) {
```

Comment:
Similar point here -- I think this works, but ideally there would be a `if (forceNewIssue)` check much earlier, and the line `currentBuildResult = getPreviousCompletedBuild()` doesn't even execute in that if block.  (We should be able to just pass a hard-coded `null` to currentBuildResultFailure if forceNewIssue is true.)